### PR TITLE
Fix checks when printing RS Instructions

### PR DESCRIPTION
**Pull request by dchopra001**
The getSourceImmediate() API is being used incorrectly inside `TR_Debug::print(TR::FILE*,TR::S390RSInstruction*)`. The purpose of the query here is to check if _sourceImmediate is set. However, if it is set to 0, then the query evaluates as false. This can cause errors and crashes due to printing. This commit adds a new query to check if `_sourceImmediate` has been set, and cleans up the print function to correctly use the `hasMask()` query as well.

Signed-off-by: Dhruv Chopra <Dhruv.C.Chopra@ibm.com>

### CODE REVIEW

File: compiler/z/codegen/S390Instruction.hpp [hunk 1]
```diff
@@ -2551,6 +2551,7 @@ class S390RSInstruction : public TR::S390RegInstruction
    int16_t                 _sourceImmediate;
    int8_t                  _maskImmediate;
    bool                    _hasMask;
+   bool                    _hasImmediate;
```
Comment (fjeremic) [inline compiler/z/codegen/S390Instruction.hpp]: Would it be possible to add Doxygen documentation for the newly added field and API?

**Comment by fjeremic** [comment]
@dchopra001 something seems off with your commit authorship. You may want to check that your GitHub profile matches the email you specified in your commit. You may wish to add alternative emails in your profile.

**Comment by dchopra001** [comment]
@fjeremic I've added documentation (4769ec2) and done some renaming (c5cb3a6). PTAL.

File: compiler/z/codegen/S390Instruction.hpp [hunk 2]
```diff
@@ -2550,7 +2550,10 @@ class S390RSInstruction : public TR::S390RegInstruction
 
    int16_t                 _sourceImmediate;
    int8_t                  _maskImmediate;
-   bool                    _hasMask;
+   /// _hasMaskImmediate evaluates to true if a value for the _maskImmediate field has been specified, false otherwise
+   bool                    _hasMaskImmediate;
```
Comment (fjeremic) [inline compiler/z/codegen/S390Instruction.hpp]: I don't think this is a very descriptive comment. The name of the field already implies relation to `_maskImmediate`. The comment doesn't give any indication as to _why_ we have the two `_sourceImmediate` and `_maskImmediate` fields. Perhaps a better comment may describe why we make the distinction:

```
/// Determines whether a mask value was specified during the construction of this object, indicating an RS instruction of RS-b format where the third operand is a mask
bool _hasMaskImmediate;

/// Determines whether an immediate value was specified during the construction of this object, indicating an RS instruction of RS-a format where the third operand is a memory reference interpreted as an immediate value.
bool _hasSourceImmediate;

Comment (dchopra001) [reply compiler/z/codegen/S390Instruction.hpp]: Updated here: 387883a

**Comment by fjeremic** [comment]
@genie-omr build all

**Pull request merged**